ViewModel/HyvaGridInterface.php

```diff
@@ -19,10 +19,11 @@ public function getColumnDefinitions(): array;
     public function getAllColumnDefinitions(): array;
 
     /**
-     * @param SearchCriteriaInterface|null $searchCriteria
      * @return HyvaGrid\RowInterface[]
      */
-    public function getRows(?SearchCriteriaInterface $searchCriteria): array;
+    public function getRows(): array;
+
+    public function getRowsForSearchCriteria(SearchCriteriaInterface $searchCriteria): array;
```

Comment:
Wie wäre es ein neues Interface `HyvaGridExportInterface` mit den Methoden `getSearchCriteria` und `getRowsForSearchCriteria` anzulegen?
Das GridViewModel kann dann sowohl `HyvaGridInterface` als auch `HyvaGridExportInterface` implementieren.
Dadurch kommt keine neue Methode auf das `HyvaGridInterface` die in dem Template gar nicht gebraucht wird, und der Source Iterator hat auch ein Eindeutiges Interface mit genau den zwei Methoden die es braucht.